ui/src/app/google-chart/donut-chart/donut-chart.component.ts

```diff
@@ -0,0 +1,49 @@
+import {Component, Input, OnDestroy, OnInit} from '@angular/core';
+
+import {GoogleChartService} from 'app/google-chart/google-chart.service';
+import {DataService} from 'app/service/data.service';
+import {TagStats} from 'compiled_proto/src/proto/tagpost_pb';
+
+@Component({
+  selector: 'app-donut-chart',
+  templateUrl: './donut-chart.component.html',
+  styleUrls: ['./donut-chart.component.sass']
+})
+/**
+ * A donut chart component to visualize TagStats
+ */
+export class DonutChartComponent implements OnInit {
+
+  private library;
+  @Input() tagStats: TagStats; // Input passed from ThreadDetailComponent
+
+  constructor(private chartService: GoogleChartService) {
+    this.library = this.chartService.getGoogle();
+    this.library.charts.load('current', { packages: ['corechart'] });
```

Comment:
Ideally this should be loaded only once.

Not sure what's the best practice in Angular to load an asynchronous resource. Maybe wrap it in a `Promise` and return it from `GoogleChartService`?